packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp

```diff
@@ -0,0 +1,37 @@
+import "@azure-tools/cadl-ranch-expect";
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@azure-tools/typespec-client-generator-core";
+import "./identity.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.ResourceManager;
+using TypeSpec.OpenAPI;
+
+@armProviderNamespace
+@service
+@versioned(Versions)
+@doc("Arm Common Types Managed Identity Provider management API.")
+namespace Azure.Arm.Models.CommonTypes.ManagedIdentity;
+
+@doc("Azure API versions.")
+enum Versions {
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+  @doc("Preview API version 2023-12-01-preview.")
+  v2023_12_01_preview: "2023-12-01-preview",
+}
+
+union ProvisioningState {
```

Comment:
Is it required in this test?

Reply:
The union `ProvisioningState` has been removed.